@ret2libc failures are unrelated:
- https://github.com/rizinorg/rizin/issues/1344

```
c  -o librz/type/librz_type.so.0.3.0-git librz/type/librz_type.so.0.3.0-git.p/base.c.o librz/type/librz_type.so.0.3.0-git.p/format.c.o librz/type/librz_type.so.0.3.0-git.p/function.c.o librz/type/librz_type.so.0.3.0-git.p/helpers.c.o librz/type/librz_type.so.0.3.0-git.p/path.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_functions.c.o librz/type/librz_type.so.0.3.0-git.p/serialize_types.c.o librz/type/librz_type.so.0.3.0-git.p/type.c.o librz/type/librz_type.so.0.3.0-git.p/typeclass.c.o librz/type/librz_type.so.0.3.0-git.p/parser_c_cpp_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_parser.c.o librz/type/librz_type.so.0.3.0-git.p/parser_types_storage.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,librz_type.so.0.3.0-git '-Wl,-rpath,$ORIGIN/../util' -Wl,-rpath-link,/__w/rizin/rizin/rizin/build/librz/util librz/util/librz_util.so.0.3.0-git -Wl,--start-group subprojects/tree-sitter-0.19.4/libtree_sitter.a subprojects/tree-sitter-c/libtree-sitter-c.a -Wl,--end-group
subprojects/tree-sitter-0.19.4/libtree_sitter.a(lib_src_lib.c.o): In function `clock_now':
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
/__w/rizin/rizin/rizin/build/../subprojects/tree-sitter-0.19.4/lib/src/././clock.h:73: undefined reference to `clock_gettime'
collect2: error: ld returned 1 exit status
[941/1667] Compiling C object librz/analysis/librz_analysis.so.0.3.0-git.p/data.c.o
[942/1667] Compiling C object librz/asm/librz_asm.so.0.3.0-git.p/arch_hexagon_hexagon_disas.c.o
```

--- a/librz/include/rz_types.h
+++ b/librz/include/rz_types.h
@@ -86,10 +86,6 @@ typedef enum {
 #define RZ_PERM_ACCESS 32
 #define RZ_PERM_CREAT  64
 
-// HACK to fix capstone-android-mips build
-#undef mips
-#define mips mips
-
 #if defined(__powerpc) || defined(__powerpc__)
 #undef __POWERPC__
 #define __POWERPC__ 1
@@ -181,14 +177,6 @@ typedef int socklen_t;
 #define FUNC_ATTR_USED                  __attribute__((used))
 #define FUNC_ATTR_WARN_UNUSED_RESULT    __attribute__((warn_unused_result))
 #define FUNC_ATTR_ALWAYS_INLINE         __attribute__((always_inline))
-
-#ifdef __clang__
-// clang only
-#elif defined(__INTEL_COMPILER)
-// intel only
-#else
-// gcc only
-#endif
 #else
 #define FUNC_ATTR_MALLOC
 #define FUNC_ATTR_ALLOC_SIZE(x)

--- a/librz/include/rz_types.h
+++ b/librz/include/rz_types.h
@@ -95,40 +95,9 @@ typedef enum {
 #define __POWERPC__ 1
 #endif
 
-#if __IPHONE_8_0 && defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
-#define LIBC_HAVE_SYSTEM 0
-#else
-#define LIBC_HAVE_SYSTEM 1
-#endif
-
-#if APPLE_SDK_IPHONEOS || APPLE_SDK_APPLETVOS || APPLE_SDK_WATCHOS || APPLE_SDK_APPLETVSIMULATOR || APPLE_SDK_WATCHSIMULATOR
-#define LIBC_HAVE_PTRACE 0
-#else
-#define LIBC_HAVE_PTRACE 1
-#endif
-
-#if HAVE_FORK
-#define LIBC_HAVE_FORK 1
-#else
-#define LIBC_HAVE_FORK 0
-#endif
-
 #if defined(__OpenBSD__)
 #include <sys/param.h>
 #undef MAXCOMLEN /* redefined in zipint.h */
-#endif
-
-/* release >= 5.9 */
-#if __OpenBSD__ && OpenBSD >= 201605
-#define LIBC_HAVE_PLEDGE 1
-#else
-#define LIBC_HAVE_PLEDGE 0
-#endif
-
-#if __sun
-#define LIBC_HAVE_PRIV_SET 1
-#else
-#define LIBC_HAVE_PRIV_SET 0
 #endif
 
 #ifdef __GNUC__

--- a/librz/include/rz_types_base.h
+++ b/librz/include/rz_types_base.h
@@ -219,8 +219,6 @@ typedef struct _utX {
 #endif
 
 #if APPLE_SDK_IPHONESIMULATOR
-#undef LIBC_HAVE_FORK
-#define LIBC_HAVE_FORK 0
 #undef DEBUGGER
 #define DEBUGGER 0
 #endif

--- a/librz/main/rz-agent.c
+++ b/librz/main/rz-agent.c
@@ -95,7 +95,7 @@ RZ_API int rz_main_rz_agent(int argc, const char **argv) {
 	memorystatus_control(MEMORYSTATUS_CMD_SET_JETSAM_TASK_LIMIT, getpid(), 256, NULL, 0);
 #endif
 	if (dodaemon) {
-#if LIBC_HAVE_FORK
+#if HAVE_FORK
 		int pid = rz_sys_fork();
 		if (pid > 0) {
 			printf("%d\n", pid);

--- a/librz/socket/run.c
+++ b/librz/socket/run.c
@@ -14,7 +14,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
-#if __APPLE__ && LIBC_HAVE_FORK
+#if __APPLE__ && HAVE_FORK
 #if !__POWERPC__
 #include <spawn.h>
 #endif
@@ -62,7 +62,7 @@
 #undef HAVE_PTY
 #define HAVE_PTY 0
 #else
-#define HAVE_PTY __UNIX__ && !__ANDROID__ && LIBC_HAVE_FORK && !__sun
+#define HAVE_PTY __UNIX__ && !__ANDROID__ && HAVE_FORK && !__sun
 #endif
 
 #if HAVE_PTY
@@ -1064,12 +1064,12 @@ RZ_API int rz_run_config_env(RzRunProfile *p) {
 
 // NOTE: return value is like in unix return code (0 = ok, 1 = not ok)
 RZ_API int rz_run_start(RzRunProfile *p) {
-#if LIBC_HAVE_FORK
+#if HAVE_EXECVE
 	if (p->_execve) {
 		exit(rz_sys_execv(p->_program, (char *const *)p->_args));
 	}
 #endif
-#if __APPLE__ && !__POWERPC__ && LIBC_HAVE_FORK
+#if __APPLE__ && !__POWERPC__ && HAVE_FORK
 	posix_spawnattr_t attr = { 0 };
 	pid_t pid = -1;
 	int ret;
@@ -1245,13 +1245,12 @@ RZ_API int rz_run_start(RzRunProfile *p) {
 				}
 			}
 			setsid();
-#if !LIBC_HAVE_FORK
+#if HAVE_EXECVE
 			exit(rz_sys_execv(p->_program, (char *const *)p->_args));
 #endif
 #endif
 		}
-// TODO: must be HAVE_EXECVE
-#if LIBC_HAVE_FORK
+#if HAVE_EXECVE
 		exit(rz_sys_execv(p->_program, (char *const *)p->_args));
 #endif
 	}

--- a/librz/socket/socket_proc.c
+++ b/librz/socket/socket_proc.c
@@ -15,7 +15,7 @@
 #define BUFFER_SIZE 4096
 
 RZ_API struct rz_socket_proc_t *rz_socket_proc_open(char *const argv[]) {
-#if __UNIX__ && LIBC_HAVE_FORK
+#if __UNIX__ && HAVE_FORK
 	RzSocketProc *sp = RZ_NEW(RzSocketProc);
 
 	if (!sp) {

